src/main/java/org/myrobotlab/service/ServoMixer.java

```diff
@@ -9,75 +9,188 @@
 import java.util.TreeSet;
 
 import org.myrobotlab.codec.CodecUtils;
+import org.myrobotlab.framework.Registration;
 import org.myrobotlab.framework.Service;
 import org.myrobotlab.framework.interfaces.Attachable;
 import org.myrobotlab.framework.interfaces.ServiceInterface;
+import org.myrobotlab.io.FileIO;
 import org.myrobotlab.kinematics.Pose;
-import org.myrobotlab.kinematics.PoseSequence;
-import org.myrobotlab.kinematics.Sequence;
+import org.myrobotlab.kinematics.PoseMove;
+import org.myrobotlab.kinematics.GesturePart;
+import org.myrobotlab.kinematics.Gesture;
 import org.myrobotlab.logging.LoggerFactory;
 import org.myrobotlab.logging.LoggingFactory;
+import org.myrobotlab.service.config.ServoMixerConfig;
+import org.myrobotlab.service.interfaces.ServiceLifeCycleListener;
 import org.myrobotlab.service.interfaces.ServoControl;
 import org.slf4j.Logger;
 
 /**
  * ServoMixer - a service which can control multiple servos. The position of the
  * servos can be saved into multiple poses, and the poses in turn can be saved
- * in a sequence file to play back later.
+ * in a gesture file to play back later.
+ * 
+ * @author GroG, kwatters, others...
+ * 
  */
-public class ServoMixer extends Service {
+public class ServoMixer extends Service implements ServiceLifeCycleListener {
 
-  public final static Logger log = LoggerFactory.getLogger(InMoov2.class);
+  public class Player implements Runnable {
+    String gestureName = null;
+    int poseIndex = 0;
+    boolean running = false;
+    Gesture runningGesture = null;
+    Thread thread = null;
 
-  private static final long serialVersionUID = 1L;
+    private void play() {
+      if (runningGesture.parts != null) {
+        invoke("publishGestureStarted", gestureName);
+        for (int i = 0; i < runningGesture.parts.size(); ++i) {
+          if (!running) {
+            break;
+          }
+          GesturePart sp = runningGesture.parts.get(i);
+          invoke("publishPlayingGesturePart", sp);
+          invoke("publishPlayingGesturePartIndex", i);
+          switch(sp.type) {

```

Comment:
not sure I know what feature your talking about .. :thinking:

Reply:
A simple primer article on pattern matching and switch:

https://www.baeldung.com/java-switch-pattern-matching

Essentially, it would let you get rid of the `sp.type` entirely and switch on the type of the value instead. so:

```java
switch(sp.value) {
    case Integer delay -> sleep(delay);
    case String speech -> invoke("publishText", speech);
    // Etc
}
```

It also enforces exhaustive checking for sealed types (a different feature available in Java 17+ that lets you restrict the subclasses/implementors of an interface or class) so you can guarantee you don't miss a case

Reply:
interesting